### ISSUE TITLE
AntiForgeryToken is broken - Server cannot append header after HTTP headers have been sent.

### ISSUE THREAD

**Issue by gauravbhavsar** [aspnet/Antiforgery#14]
I am implementing AntiForgeryToken my login form.

When I run project, its call Account/Login Get mothod and return view() i.e login.cshtml

In login.cshtml view page, I have 

    @using (Html.BeginForm("Login", "Account", new { ReturnUrl = ViewBag.ReturnUrl }, FormMethod.Post, new { @class = "form1" }))
    {
        @Html.AntiForgeryToken(); // throw exception 
        <label class="username_style" for="UserName"></label>
        @Html.TextBoxFor(m => m.UserName, new { placeholder = "UserName", autofocus = true })

        <label class="password_style" for="Password"></label>
        @Html.TextBoxFor(m => m.Password, new { type = "password", placeholder = "Password" })

        <p class="msg error">@Html.ValidationSummary(false, "", new { @class = "msg error" })</p>
        <input type="submit" name="login" value="Login" id="btnLogin" />
        <p><a href="javascript:void(0)" class="elink">Forgot Password?</a></p>
    }

When server return login.cshtml page I got this exception 

I am implementing AntiForgeryToken my login form.

When I run project, its call Account/Login Get mothod and return view() i.e login.cshtml

In login.cshtml view page, I have 

    @using (Html.BeginForm("Login", "Account", new { ReturnUrl = ViewBag.ReturnUrl }, FormMethod.Post, new { @class = "form1" }))
    {
        @Html.AntiForgeryToken(); // throw exception 
        <label class="username_style" for="UserName"></label>
        @Html.TextBoxFor(m => m.UserName, new { placeholder = "UserName", autofocus = true })

        <label class="password_style" for="Password"></label>
        @Html.TextBoxFor(m => m.Password, new { type = "password", placeholder = "Password" })

        <p class="msg error">@Html.ValidationSummary(false, "", new { @class = "msg error" })</p>
        <input type="submit" name="login" value="Login" id="btnLogin" />
        <p><a href="javascript:void(0)" class="elink">Forgot Password?</a></p>
    }

When server return login.cshtml page

**Server cannot append header after HTTP headers have been sent.**

I can not figure out how to solve this error.

Thank you,

**Comment by rynowak** [comment]
This message means that something is happening that's causing the server to start sending data back to the client before you add the cookie for antiforgery. Do you have a `Flush()` in your layout page?

If you have a small isolated repro of this you put in a repository we have access to we can help you troubleshoot the problem.

**Comment by Eilon** [comment]
Closing until we have a repro. This is almost certainly what @rynowak stated. If you can share a repro project we can take another look at this.

**Comment by gauravbhavsar** [comment]
Sorry for late reply, I created new project and I am not able to regenerate an issue, may be @rynowak  is right that something is happening that's causing the server to start seding data back to the client. So I created new new project and transfer all login and code to new created project and from now I am not getting such error. And also cliam Antiforgery Token in global.

    AntiForgeryConfig.UniqueClaimTypeIdentifier = ClaimTypes.NameIdentifier;
